build/Targets/Versions.props

```diff
@@ -41,6 +41,10 @@
     <!-- NuGet version -->
     <NuGetReleaseVersion>$(RoslynFileVersionBase)</NuGetReleaseVersion>
     <NuGetPerBuildPreReleaseVersion>$(NuGetPreReleaseVersion)-$(BuildNumberFiveDigitDateStamp)-$(BuildNumberBuildOfTheDayPadded)</NuGetPerBuildPreReleaseVersion>
+    <NuGetPackageRoot Condition="'$(NuGetPackageRoot)' == ''">$(NUGET_PACKAGES)</NuGetPackageRoot> <!-- Respect environment variable if set -->
```

Comment:
A considerable portion of the values defined in Version.props, and by extension Dependencies.props, are NuGet packages.  Not having the root for which to find them reduced the values of these files.  

Importing Settings.props is a bit of a heavier hand is it brings along a lot of properties which are specific to building.  There are a number of build files that are just tooling that can't always import these values.  Hence I broke it up this way. 

I've tried to document the intent of these files in the following README.md.  Happy to update if you feel necessary

https://github.com/dotnet/roslyn/blob/master/build/Targets/README.md